_studio/mfx_lib/shared/src/mfx_h264_fei_vaapi.cpp

```diff
@@ -1775,17 +1774,29 @@ mfxStatus VAAPIFEIPAKEncoder::Destroy()
 {
     MFX_AUTO_LTRACE(MFX_TRACE_LEVEL_HOTSPOTS, "FEI::PAK::Destroy");
 
-    mfxStatus sts = MFX_ERR_NONE;
-
-    MFX_DESTROY_VABUFFER(m_statParamsId,     m_vaDisplay);
-    MFX_DESTROY_VABUFFER(m_statMVId,         m_vaDisplay);
-    MFX_DESTROY_VABUFFER(m_statOutId,        m_vaDisplay);
-    MFX_DESTROY_VABUFFER(m_codedBufferId[0], m_vaDisplay);
-    MFX_DESTROY_VABUFFER(m_codedBufferId[1], m_vaDisplay);
-
-    sts = VAAPIEncoder::Destroy();
+#if defined(_DEBUG)
+    mfxStatus sts;
+    sts =
+#endif
```

Comment:
I don't like this section. If someone needs to see variable in debug - let him patch according and rebuild. I vote for dropping these if-defs.

Reply:
@dvrogozh, +1
Anyway, stepping into `CheckAndDestroyVAbuffer` is also an option.

Reply:
@dmitry-gurulev @dvrogozh 
what about following solution:
```
mfxStatus sts = CheckAndDestroyVAbuffer(...);
MFX_STS_TRACE(sts);
```

Reply:
Please don't introduce __DEBUG stuff. MFX_STS_TRACE(sts); a bit better still ideally we should actually handle errors, not just log them.

Reply:
@onabiull this is ```Close``` function, i.e. = destructor. What is your suggestion here?

Reply:
As a short-term logging is enough. But considering possible fail sources - corrupted heap, corrupted VA context, probably we should just fail here.

Reply:
@dvrogozh @onabiull @dmitry-gurulev please see new version